### ISSUE TITLE
was collected before with the same name and label values

### ISSUE THREAD

**Issue by nri-pl** [timonwong/uwsgi_exporter#22]
$ uwsgi_exporter -web.listen-address :9100 -log.level debug -stats.uri unix:///tmp/uwsgi-stats.socket
```
INFO[0000] Starting uwsgi_exporter (version=0.6.0, branch=HEAD, revision=d848c556c1aa103eb058705d96df3aefcb4ccdb3)  source="uwsgi_exporter.go:37"
INFO[0000] Build context (go=go1.8.5, user=root@3f72bd5c99ad, date=20171223-06:35:18)  source="uwsgi_exporter.go:38"
INFO[0000] Listening on :9100                            source="uwsgi_exporter.go:56"
DEBU[0004] OK: scrape successful after 0.000722s.        source="exporter.go:174"
DEBU[0012] OK: scrape successful after 0.000568s.        source="exporter.go:174"
```

$ curl 127.0.0.1:9100/metrics
```
An error has occurred during metrics collection:

4 error(s) occurred:
* collected metric uwsgi_socket_queue_length label:<name:"name" value:":3000" > label:<name:"proto" value:"uwsgi" > label:<name:"stats_uri" value:"unix:///tmp/uwsgi-stats.socket" > gauge:<value:0 >  was collected before with the same name and label values
* collected metric uwsgi_socket_max_queue_length label:<name:"name" value:":3000" > label:<name:"proto" value:"uwsgi" > label:<name:"stats_uri" value:"unix:///tmp/uwsgi-stats.socket" > gauge:<value:0 >  was collected before with the same name and label values
* collected metric uwsgi_socket_shared label:<name:"name" value:":3000" > label:<name:"proto" value:"uwsgi" > label:<name:"stats_uri" value:"unix:///tmp/uwsgi-stats.socket" > gauge:<value:0 >  was collected before with the same name and label values
* collected metric uwsgi_socket_can_offload label:<name:"name" value:":3000" > label:<name:"proto" value:"uwsgi" > label:<name:"stats_uri" value:"unix:///tmp/uwsgi-stats.socket" > gauge:<value:0 >  was collected before with the same name and label values

```

What am I doing wrong?

$ uwsgi_exporter -version
```
uwsgi_exporter, version 0.6.0 (branch: HEAD, revision: d848c556c1aa103eb058705d96df3aefcb4ccdb3)
  build user:       root@3f72bd5c99ad
  build date:       20171223-06:35:18
  go version:       go1.8.5
```

**Comment by timonwong** [comment]
Hi, you probably hit a bug.
Can you give me the stats result using this command `nc -U /tmp/uwsgi-stats.socket ?

**Comment by nri-pl** [comment]
```
{
	"version":"2.0.15",
	"listen_queue":0,
	"listen_queue_errors":0,
	"signal_queue":0,
	"load":0,
	"pid":37,
	"uid":65534,
	"gid":65534,
	"cwd":"/opt",
	"locks":[
		{
			"user 0":0
		},
		{
			"signal":0
		},
		{
			"filemon":0
		},
		{
			"timer":0
		},
		{
			"rbtimer":0
		},
		{
			"cron":0
		},
		{
			"thunder":38
		},
		{
			"rpc":0
		},
		{
			"snmp":0
		},
		{
			"cache_business_service_cache":0
		}
	],
	"caches":[
		{
			"name":"business_service_cache",
			"hash":"djb33x",
			"hashsize":65536,
			"keysize":2048,
			"max_items":1000,
			"blocks":1000,
			"blocksize":65536,
			"items":0,
			"hits":0,
			"miss":0,
			"full":0,
			"last_modified_at":0
		}
	],
	"sockets":[
		{
			"name":":3000",
			"proto":"uwsgi",
			"queue":0,
			"max_queue":120,
			"shared":0,
			"can_offload":0
		},
		{
			"name":":3000",
			"proto":"uwsgi",
			"queue":0,
			"max_queue":0,
			"shared":0,
			"can_offload":0
		}
	],
	"workers":[
		{
			"id":1,
			"pid":38,
			"accepting":1,
			"requests":2,
			"delta_requests":2,
			"exceptions":0,
			"harakiri_count":0,
			"signals":0,
			"signal_queue":0,
			"status":"idle",
			"rss":0,
			"vsz":0,
			"running_time":363987,
			"last_spawn":1518071880,
			"respawn_count":1,
			"tx":18103,
			"avg_rt":137610,
			"apps":[
				{
					"id":0,
					"modifier1":0,
					"mountpoint":"",
					"startup_time":3,
					"requests":2,
					"exceptions":0,
					"chdir":""
				}
			],
			"cores":[
				{
					"id":0,
					"requests":2,
					"static_requests":0,
					"routed_requests":0,
					"offloaded_requests":0,
					"write_errors":0,
					"read_errors":0,
					"in_request":0,
					"vars":[

					],
					"req_info":					{

					}
				}
			]
		},
		{
			"id":2,
			"pid":39,
			"accepting":1,
			"requests":2,
			"delta_requests":2,
			"exceptions":0,
			"harakiri_count":0,
			"signals":0,
			"signal_queue":0,
			"status":"idle",
			"rss":0,
			"vsz":0,
			"running_time":83588,
			"last_spawn":1518071880,
			"respawn_count":1,
			"tx":23643,
			"avg_rt":32558,
			"apps":[
				{
					"id":0,
					"modifier1":0,
					"mountpoint":"",
					"startup_time":3,
					"requests":2,
					"exceptions":0,
					"chdir":""
				}
			],
			"cores":[
				{
					"id":0,
					"requests":2,
					"static_requests":0,
					"routed_requests":0,
					"offloaded_requests":0,
					"write_errors":0,
					"read_errors":0,
					"in_request":0,
					"vars":[

					],
					"req_info":					{

					}
				}
			]
		}
	]
}
```

**Comment by timonwong** [comment]
The main problem is there is two "sockets" with same `"name"` and `"proto"`. So it's probably caused by multiple `--socket` arguments passed with same value.
I'm not sure it's legal here, I'll try to find out.

**Comment by nri-pl** [comment]
It also seemed strange to me.
In uWSGI conf.ini I have only one "socket = :3000"

```
[uWSGI] getting INI configuration from /opt/config/api.ini
*** Starting uWSGI 2.0.15 (64bit) on [Thu Feb  8 06:38:00 2018] ***
compiled with version: 6.4.0 on 04 December 2017 14:09:38
os: Linux-4.13.16-coreos-r1 #1 SMP Thu Nov 30 16:58:14 UTC 2017
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 10
current working directory: /opt
detected binary path: /usr/bin/uwsgi
chdir() to /opt/
your processes number limit is 120504
your memory page size is 4096 bytes
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers ***
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: enabled
*** Cache "business_service_cache" initialized: 64MB (key: 2136 bytes, keys: 2136000 bytes, data: 65536000 bytes, bitmap: 125 bytes) preallocated ***
uwsgi socket 0 bound to TCP address :3000 fd 4
Python version: 2.7.14 (default, Nov  9 2017, 20:04:23)  [GCC 6.4.0]
Python main interpreter initialized at 0x55b5f8787980
python threads support enabled
your server socket listen backlog is limited to 120 connections
your mercy for graceful operations on workers is 8 seconds
mapped 296016 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
spawned uWSGI master process (pid: 37)
spawned uWSGI worker 1 (pid: 38, cores: 1)
spawned uWSGI worker 2 (pid: 39, cores: 1)
cache sweeper thread enabled
```

This looks like some error in UWSGI.
But maybe the exporter should deal with it.

**Comment by timonwong** [comment]
I'll add a workaround for this

**Comment by timonwong** [comment]
@nri-pl This issue is addressed in v0.7.0
Please note that you now need to use double dash in flags, for example (`--stats.uri`)

**Comment by nri-pl** [comment]
Thx
